src/Frontend/src/views/ThroughputReportView.spec.ts

```diff
@@ -0,0 +1,126 @@
+import Toast from "vue-toastification";
+import { serviceControlWithThroughput } from "@/views/throughputreport/serviceControlWithThroughput";
+
+describe("EndpointsView tests", () => {
+  const serviceControlInstanceUrl = window.defaultConfig.service_control_url;
+
+  async function setup() {
+    const driver = makeDriverForTests();
+
+    await driver.setUp(serviceControlWithThroughput);
+
+    return driver;
+  }
+
+  async function renderComponent(transport: Transport = Transport.MSMQ, preSetup: (driver: Driver) => Promise<void> = () => Promise.resolve()) {
+    disableMonitoring();
+
+    const driver = await setup();
+    await preSetup(driver);
+
+    useServiceControlUrls();
+    await useServiceControl();
+    const el = document.createElement("div");
+    el.id = "modalDisplay";
+    document.body.appendChild(el);
+
+    const { debug } = render(ThroughputReportView, {
+      container: document.body,
+      global: {
+        stubs: {
+          RouterLink: RouterLinkStub,
+        },
+        plugins: [makeRouter(), Toast, createTestingPinia({ stubActions: false })],
+      },
+    });
+    await flushPromises();
+
+    return { debug, driver };
+  }
+
+  describe("when minimum requirements", () => {
+    test("are met", async () => {
+      await renderComponent(Transport.AmazonSQS, async (driver) => {
+        await driver.setUp(precondition.hasLicensingReportAvailable());
+      });
+
+      expect(screen.queryByText(/the minimum version of servicecontrol required to enable the usage feature is/i)).not.toBeInTheDocument();
+    });
+
+    test("are not met, requirements warning is displayed", async () => {
+      await renderComponent(Transport.AmazonSQS, async (driver) => {
+        await driver.setUp(precondition.hasServiceControlMainInstance("1.0.0"));
+        await driver.setUp(precondition.hasLicensingReportAvailable());
+      });
+
+      expect(screen.getByText(/the minimum version of servicecontrol required to enable the usage feature is/i)).toBeInTheDocument();
+    });
+  });
+
+  describe("when report", () => {
+    test("is available", async (driver) => {
+      await renderComponent(Transport.AmazonSQS, async (driver) => {
+        await driver.setUp(precondition.hasLicensingReportAvailable());
+      });
+      expect(screen.getByRole("button", { name: /Download Report/i })).toBeEnabled();
+    });
+
+    test("is unavailable", async (driver) => {
+      const reason = "report testing that is not available";
+      await renderComponent(Transport.AmazonSQS, async (driver) => {
+        await driver.setUp(precondition.hasLicensingReportAvailable({ report_can_be_generated: false, reason: reason }));
+      });
+
+      expect(screen.getByRole("button", { name: /Download Report/i })).toBeDisabled();
+      expect(screen.getByText(reason)).toBeInTheDocument();
+    });
+  });
+
+  describe("when download report is clicked", () => {
+    test("and no warnings, download happens", async (driver) => {
```

Comment:
Could we also include tests that describe the expected content of the file for a given set of usage conditions?

Reply:
I suppose that is already tested on the backend.

Reply:
yes the content is a backend responsibility